### PR TITLE
i18n - TypeScript

### PR DESCRIPTION
**Pull request by johndiiorio**
Adds French translations

### CODE REVIEW

**Review by buip — APPROVED** [review]
The server side looks good too me

**Review by sandeepacharya464 — APPROVED** [review]
Great work John!  To me, it looks great.

File: .babelrc [hunk 1]
```diff
@@ -3,5 +3,13 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 {
 	"presets": ["env", "react"],
-	"plugins": ["transform-object-rest-spread"]
+	"plugins": [
+		"transform-object-rest-spread", 
+		[
+			"react-intl", {
+				"messagesDir": "./src/client/app/translations/messages",
```
Comment (simonbtomlinson) [inline .babelrc]: If we remove this we shouldn't need to worry about generated files at all.

File: .babelrc [hunk 2]
```diff
@@ -3,5 +3,13 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 {
 	"presets": ["env", "react"],
-	"plugins": ["transform-object-rest-spread"]
+	"plugins": [
+		"transform-object-rest-spread",
```
Comment (simonbtomlinson) [inline .babelrc]: Do we still need this line post-typescript? @LeoTindall do you know?

File: .gitignore [hunk 3]
```diff
@@ -31,3 +31,6 @@ postgres-data
 # Text files
 *.txt
 !License.txt
+
+# Generated translations
```
Comment (simonbtomlinson) [inline .gitignore]: We shouldn't need this if we remove the line in the baberc file.

File: src/client/app/utils/translate.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import store from '../index';
+import { addLocaleData, IntlProvider, defineMessages } from 'react-intl';
+import * as en from 'react-intl/locale-data/en';
+import * as fr from 'react-intl/locale-data/fr';
+import * as localeData from '../translations/locales/data.json';
+
+export default function translate(messageID: string): string {
```
Comment (simonbtomlinson) [inline src/client/app/utils/translate.ts]: This feels very heavyweight for something that runs each time a string gets translated. Can you cache most of this on the first run?

Comment (simonbtomlinson) [reply src/client/app/utils/translate.ts]: You also might want to export it with a shorter name like `t` so it doesn't add a ton of line length when used. Django does something similar with its translations.

**Review by simonbtomlinson — CHANGES_REQUESTED** [review]

File: src/client/app/utils/translate.ts (hunk 4 above)
Comment (johndiiorio) [reply src/client/app/utils/translate.ts]: `t` is frequently used throughout Redux to denote types

**Review by simonbtomlinson — APPROVED** [review]

**Pull request merged**